### PR TITLE
CI: dependabot config for major versions & minor caf

### PR DESCRIPTION
**Pull request by MattBuckley-TfN**
# Changes

Updated dependabot config to run at the end of the month (28th) and to check GitHub actions versions. Also, switched to only work for major package versions and minors for caf packages.

# Tasks

- [x] Have unittests been added (testing individual functions and their edge cases) - NA
- [x] Has documentation (including user guide) been updated - NA
- [x] Have new dependencies been added (or changed) in relevant `requirements.txt` - NA
- [ ] Code linting, tests and other workflows pass

### CODE REVIEW

File: {{ cookiecutter.project_slug }}/.github/dependabot.yml [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
-
 version: 2
 updates:
   - package-ecosystem: "pip"
     directory: "/" # Location of package manifests
     schedule:
-      interval: "weekly"
-
+      interval: "cron"
+      cronjob: "0 0 28 * *"
```
[the remark this answers is not in the document]
Comment (MattBuckley-TfN) [reply {{ cookiecutter.project_slug }}/.github/dependabot.yml:7]: Incorrect

File: {{ cookiecutter.project_slug }}/.github/dependabot.yml [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
           - "version-update:semver-minor"
-
     # If you only want security update PRs, set this to 0
     open-pull-requests-limit: 3
+
+  - package-ecosystem: "github-actions"
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "cron"
+      cronjob: "0 0 28 * *"
```
[the remark this answers is not in the document]
Comment (MattBuckley-TfN) [reply {{ cookiecutter.project_slug }}/.github/dependabot.yml:24]: Incorrect

File: {{ cookiecutter.project_slug }}/.github/dependabot.yml [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
+    open-pull-requests-limit: 1
```
[the remark this answers is not in the document]
Comment (MattBuckley-TfN) [reply {{ cookiecutter.project_slug }}/.github/dependabot.yml:25]: GitHub actions don't have separate major or minor versions

**Review by BenTaylor-TfN — APPROVED** [review]
